### PR TITLE
4.x - Middleware queue updates

### PR DESCRIPTION
**Pull request by ADmad**
- `MiddlewareQueue` now implements `SeekableIterator` to avoid having to maintain iteration index in `Runner`.
- Using a separate array to maintain resolved middleware instances seemed unnecessary, so I got rid of `MiddlewareQueue::$middlewares` and `MiddlewareQueue::$queue` is directly updated now.

### CODE REVIEW

**Comment by ADmad** [comment]
@markstory Any feedback?

File: src/Http/Runner.php [hunk 1]
```diff
@@ -72,9 +65,10 @@ public function run(
      */
     public function handle(ServerRequestInterface $request): ResponseInterface
     {
-        $middleware = $this->queue->get($this->index++);
+        if ($this->queue->valid()) {
+            $middleware = $this->queue->current();
+            $this->queue->next();
```
Comment (markstory) [inline src/Http/Runner.php]: Operating an iterator like this is a bit clunky would `yield` be usable here?

Comment (ADmad) [reply src/Http/Runner.php]: Nope it isn't.

Runners in other libs I check also traverse the queue manually. `Relay` use `current()` and `next()` functions on it's queue which is an array. `zend-stratigility`'s queue is an `SplQueue` and it's runner uses `dequeue()` method to get next middleware.

**Comment by ADmad** [comment]
I also got rid of `MiddlewareQueue::get()` method. It was only used in tests. Picking middlewares from arbitrarily positions isn't required in normal usage of the queue.

File: src/Http/Runner.php (hunk 1 above)
Comment (markstory) [reply src/Http/Runner.php]: Ok. It was worth a shot.

File: src/Http/MiddlewareQueue.php [hunk 2]
```diff
@@ -55,37 +57,15 @@ public function __construct(array $middleware = [])
...
     }
 
-    /**
-     * Get the middleware at the provided index.
-     *
-     * @param int $index The index to fetch.
-     * @return \Psr\Http\Server\MiddlewareInterface|null Either the middleware or null
-     *   if the index is undefined.
-     */
-    public function get(int $index): ?MiddlewareInterface
```
Comment (markstory) [inline src/Http/MiddlewareQueue.php]: Do you think someone will be using this? We haven't deprecated this method in 3.7.

Comment (ADmad) [reply src/Http/MiddlewareQueue.php]: I don't see any reason why someone would need to get a middleware from queue themselves.

Comment (ADmad) [reply src/Http/MiddlewareQueue.php]: We can mark it deprecated in 4.0 and keep it until 5.0 but I really don't see the need.

Comment (markstory) [reply src/Http/MiddlewareQueue.php]: Ok. Let's remove it and see what happens.  :smile;

Comment (ADmad) [reply src/Http/MiddlewareQueue.php]: Cool, this should be good to merge then :slightly_smiling_face:.

**Pull request merged**